src/main/java/seedu/address/model/assignment/Description.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.assignment;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an Assignment's description in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDescription(String)}
+ */
+public class Description {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Assignments should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String description;
+
+    /**
+     * Constructs a {@code Description}.
+     *
+     * @param description A valid description.
+     */
+    public Description(String description) {
+        requireNonNull(description);
+        checkArgument(isValidDescription(description), MESSAGE_CONSTRAINTS);
+        this.description = description;
+    }
+
+    /**
+     * Returns true if a given string is a valid description.
+     */
+    public static boolean isValidDescription(String test) {
```

Comment:
parameter test maybe changed to some more meaningful terms like inputDescription

Reply:
I will make the necessary change, thanks!